emdgroup-liquid/liquid
# fix: add patch to fully specify resolve path in stencil output targets

# Description

This PR includes a patch to the Stencil Vue and React output targets, resolving an issue with relatively imported modules, which need to have the .js or /index.js extension to be added to the import path in order to resolve an actual path, when working with esm in environments, where a fully specified path is the default (such as in a create react app with webpack 5, which, if not ejected and changed, has [resolve.fullySpecified](https://webpack.js.org/configuration/module/#resolvefullyspecified) set to true).

Fixes #495

References:
- Issue https://github.com/ionic-team/stencil-ds-output-targets/issues/319
- PR https://github.com/ionic-team/stencil-ds-output-targets/pull/322

## Type of change


## Is it a breaking change?


# How Has This Been Tested?


In order to test this, you'll need to do the following:

1. Inside the liquid project working directory, check out the branch `fix/resolve_esm`
2. Run `yarn && yarn build && npm link`
3. Clone https://github.com/emdgroup-liquid/liquid-sandbox-cra-tailwind
4. Delete `patch-package ; ` in [this line](https://github.com/emdgroup-liquid/liquid-sandbox-cra-tailwind/blob/main/package.json#L43) in the package.json file
5. Run `npm install && npm link @emdgroup-liquid/liquid && npm start`
6. Open http://localhost:3000 in your browser

You can test the Vue bindings as well, using the same procedure as above with the [Nuxt sandbox app](https://github.com/emdgroup-liquid/liquid-sandbox-nuxt-tailwind). However, Nuxt will not be able to resolve symlinks, so you will have to copy paste the vue dist folder into the project as a workaround.

# Checklist:
